File: Excel/590 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/590 Consecutive Streaks.xlsx"
input = read_excel(path, range = "A2:B22")
test  = read_excel(path, range = "D2:F5")

process = function(df, starting_index) {
  df = df %>%
    filter(row_number() >= starting_index) %>%
    mutate(cons = consecutive_id(Result)) %>%
    filter(cons == 1) %>%
    summarise(Index = min(Index),
              Result = min(Result),
              Streak = n())
  return(df)
}

result = map_dfr(1:20, ~process(input, .x)) %>%
  filter(Streak != 1) %>%
  pivot_wider(names_from = Result, values_from = Index, values_fn = ~ paste(.x, collapse = ", ")) %>%
  select(Consecutives = Streak, everything())

all.equal(result, test, check.attributes = FALSE)
# [1] TRUE         

File: Excel/500-599/590/590 Challenge.py
import pandas as pd

path = "590 Consecutive Streaks.xlsx"
input = pd.read_excel(path, usecols="A:B", skiprows=1, nrows=21)
test = pd.read_excel(path, usecols="D:F", skiprows=1, nrows=3)
test['H'] = test['H'].astype(str)
test['T'] = test['T'].astype(str)

def consecutive_id(series):
    return (series != series.shift()).cumsum()

def process(df, starting_index):
    df = df.iloc[starting_index-1:]
    df = df.assign(cons=consecutive_id(df['Result']))
    df = df[df['cons'] == 1]
    result = {
        'Index': df['Index'].min(),
        'Result': df['Result'].min(),
        'Streak': len(df)
    }
    return pd.DataFrame([result])

result = pd.concat([process(input, i) for i in range(1, 21)])
result = result[result['Streak'] != 1]
result = result.pivot_table(index='Streak', columns='Result', values='Index', aggfunc=lambda x: ', '.join(map(str, x))).sort_values('Streak', ascending=False)
result.reset_index(inplace=True)
result.columns.name = None
result.rename(columns={'Streak': 'Consecutives'}, inplace=True)

print(result.equals(test)) # True